app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -111,13 +141,15 @@ def create_declarations(application)
     def create_outcomes(application)
       completed_declaration = application.declarations.eligible_for_outcomes(lead_provider, application.course.identifier).first
       return unless completed_declaration
+      return unless CourseGroup.joins(:courses).leadership_or_specialist.where(courses: { identifier: application.course.identifier }).exists?
 
-      ParticipantOutcomes::Create::STATES.reverse.each do |state|
```

Comment:
can we use the old version (this one) please, as otherwise we would create voided outcomes without creating a previous outcome first, also a voided outcome should only be attached to a voided declaration?